src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -204,18 +181,10 @@ public GlobalSearchBar(LibraryTabContainer tabContainer,
 
         modifierButtons.setAlignment(Pos.CENTER);
         searchField.setRight(new HBox(searchField.getRight(), modifierButtons));
-        searchField.getStyleClass().add("search-field");
+        searchField.getStyleClass().add("global-search-bar");
         searchField.setMinWidth(100);
         HBox.setHgrow(searchField, Priority.ALWAYS);
 
-        regexValidator = new FunctionBasedValidator<>(
-                searchField.textProperty(),
-                query -> !(regularExpressionButton.isSelected() && !validRegex()),
-                ValidationMessage.error(Localization.lang("Invalid regular expression")));
-        ControlsFxVisualizer visualizer = new ControlsFxVisualizer();
-        visualizer.setDecoration(new IconValidationDecorator(Pos.CENTER_LEFT));
-        Platform.runLater(() -> visualizer.initVisualization(regexValidator.getValidationStatus(), searchField));
-
```

Comment:
I've removed this validator because it is not listening for changes of the modifier buttons, an exception will be thrown in this case: write an invalid regex in the search bar, with the regex modifier disabled, then enable the regex button. will throw ```java.util.regex.PatternSyntaxException: Unclosed group near index 4
(k\)```

Also, this visualizer throws an exception in the global search window. (Open the window, enable regex modifier, write an invalid query, close the window, and open it again will throw ```java.lang.IllegalArgumentException: DialogPane@30d0220[styleClass=dialog-pane]is already inside a scene-graph and cannot be set as root```.

Reply:
Have you tried `EasyBind.combine` to combine listening for searchField text property and change of selection of the modifier button? Overall, I think we should stick to one pattern to display errors to users, unless it is necessary to implement a workaround.

Reply:
Yes, using `EasyBind#combine` works, but for the second issue, the global search window will not open again if it was closed with the visualizer.

However, we already show a message indicating that the query is invalid.
![image](https://github.com/JabRef/jabref/assets/52158423/7c7d1e1e-d152-4ef1-99b6-b951dc580c19)

With the visualizer, we only add an icon before the search bar.
![image](https://github.com/JabRef/jabref/assets/52158423/7f8c3ba0-9b56-47cf-9078-f32a66b5be8d)

Reply:
Okay, I see. If it's necessary then let's keep it.